modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PhpSlim4ServerCodegen.java

```diff
@@ -0,0 +1,368 @@
+import io.swagger.v3.oas.models.security.SecurityScheme;
+import io.swagger.v3.oas.models.servers.Server;
+import org.apache.commons.lang3.StringEscapeUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.openapitools.codegen.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.*;
+
+import static org.openapitools.codegen.utils.StringUtils.*;
+
+public class PhpSlim4ServerCodegen extends AbstractPhpCodegen {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PhpSlim4ServerCodegen.class);
+
+    public static final String USER_CLASSNAME_KEY = "userClassname";
+    public static final String PSR7_IMPLEMENTATION = "psr7Implementation";
+
+    protected String groupId = "org.openapitools";
+    protected String artifactId = "openapi-server";
+    protected String authDirName = "Auth";
+    protected String authPackage = "";
+    protected String psr7Implementation = "slim-psr7";
+    protected List<Map<String, String>> composerPackages = new ArrayList<Map<String, String>>();
+    protected List<Map<String, String>> composerDevPackages = new ArrayList<Map<String, String>>();
+
+    public PhpSlim4ServerCodegen() {
+        super();
+
+        // clear import mapping (from default generator) as slim does not use it
+        // at the moment
+        importMapping.clear();
+
+        variableNamingConvention = "camelCase";
+        artifactVersion = "1.0.0";
+        setInvokerPackage("OpenAPIServer");
+        apiPackage = invokerPackage + "\\" + apiDirName;
+        modelPackage = invokerPackage + "\\" + modelDirName;
+        authPackage = invokerPackage + "\\" + authDirName;
+        outputFolder = "generated-code" + File.separator + "slim";
+
+        modelTestTemplateFiles.put("model_test.mustache", ".php");
+        // no doc files
+        modelDocTemplateFiles.clear();
+        apiDocTemplateFiles.clear();
+
+        embeddedTemplateDir = templateDir = "php-slim4-server";
+
+        additionalProperties.put(CodegenConstants.GROUP_ID, groupId);
+        additionalProperties.put(CodegenConstants.ARTIFACT_ID, artifactId);
+
+        // override cliOptions from AbstractPhpCodegen
+        updateOption(AbstractPhpCodegen.VARIABLE_NAMING_CONVENTION, "camelCase");
+
+        // Slim 4 can use any PSR-7 implementation
+        // https://www.slimframework.com/docs/v4/concepts/value-objects.html
+        CliOption psr7Option = new CliOption(PSR7_IMPLEMENTATION,
+                "Slim 4 provides its own PSR-7 implementation so that it works out of the box. However, you are free to replace Slim’s default PSR-7 objects with a third-party implementation. Ref: https://www.slimframework.com/docs/v4/concepts/value-objects.html");
+
+        psr7Option.addEnum("slim-psr7", "Slim PSR-7 Message implementation")
+                .addEnum("nyholm-psr7", "Nyholm PSR-7 Message implementation")
```

Comment:
I'm not familiar with your Java coding style, is this indentation correct? Or it supposed to be 4 spaces instead of 8?

Reply:
No worry, we can use Intellj to format the code later